File: app/src/main/java/com/templates/kotlintemplates/views/adapters/BaseViewPagerAdapter.java
package com.templates.kotlintemplates.views.adapters;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import android.view.ViewGroup;

import java.util.HashMap;
import java.util.Map;


public abstract class BaseViewPagerAdapter extends FragmentStatePagerAdapter {

    private Map<Integer, Fragment> fragmentHashMap = new HashMap<>();

    public BaseViewPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int i) {
        return null;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        super.destroyItem(container, position, object);
        fragmentHashMap.remove(position);
    }

    @Override
    public int getCount() {
        return 0;
    }

    protected void saveFragment(Integer key, Fragment value) {
        fragmentHashMap.put(key, value);
    }

    public Fragment getFragment(int position) {
        return fragmentHashMap.get(position);
    }

}

File: app/src/main/java/com/templates/kotlintemplates/views/adapters/BaseViewPagerAdapter.kt
package com.templates.kotlintemplates.views.adapters

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter
import android.view.ViewGroup

import java.util.HashMap


abstract class BaseViewPagerAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm) {

    private val fragmentHashMap = HashMap<Int, Fragment>()

    override fun getItem(i: Int): Fragment? {
        return null
    }

    override fun destroyItem(container: ViewGroup?, position: Int, `object`: Any) {
        super.destroyItem(container, position, `object`)
        fragmentHashMap.remove(position)
    }

    override fun getCount(): Int {
        return 0
    }

    protected fun saveFragment(key: Int, value: Fragment) {
        fragmentHashMap.put(key, value)
    }

    fun getFragment(position: Int): Fragment? {
        return fragmentHashMap[position]
    }

}
